core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -556,7 +569,7 @@ private void registerBuiltinQueryParsers() {
         registerQueryParser(new CommonTermsQueryParser(), CommonTermsQueryParser.QUERY_NAME_FIELD);
         registerQueryParser(new SpanMultiTermQueryParser(), SpanMultiTermQueryParser.QUERY_NAME_FIELD);
         QueryParser<FunctionScoreQueryBuilder> functionScoreParser = (QueryParseContext c) -> FunctionScoreQueryBuilder
-                .fromXContent((String name) -> functionScoreParsers.get(name), c);
+                .fromXContent(this::lookupScoreFunctionParser, c);
```

Comment:
Would you be ok with it being a functional interface that I implement with a reference to that method in SearchModule?